src/lib_rules.ml

```diff
@@ -163,14 +163,15 @@ module Gen (P : Install_rules.Params) = struct
     Module_compilation.build_modules cctx ~js_of_ocaml ~dynlink ~dep_graphs
 
   let build_c_file (lib : Library.t) ~expander ~dir ~includes (loc, src, dst) =
+    let src = C.Source.path src in
     SC.add_rule sctx ~loc ~dir
       (Expander.expand_and_eval_set expander lib.c_flags
          ~standard:(Build.return (Context.cc_g ctx))
        >>>
        Build.run
          (* We have to execute the rule in the library directory as
             the .o is produced in the current directory *)
-         ~dir:(Path.parent_exn src)
```

Comment:
My fix doesn't seem entirely right to me but what we had before was suspicious as well. The parent of the source dir of the file isn't necessarily the library directory.